packages/grid/x-grid-modules/src/components/cell.tsx

```diff
@@ -51,7 +51,6 @@ export const Cell: React.FC<GridCellProps> = React.memo((props) => {
       role="cell"
       data-value={value}
       data-field={field}
-      data-colindex={colIndex}
```

Comment:
Duplicates with `aria-colindex` better use the a11y as the source of truth for the logic and prevent anyone to rely on arbitrary data attributes.